src/com/goide/intentions/GoMoveToStructInitializationIntention.java

```diff
@@ -0,0 +1,283 @@
+      if (fieldValue == null) continue;
+
+      GoPsiImplUtil.deleteExpressionFromAssignment(assignment, fieldReference);
+      addFieldDefinition(literalValue, getFieldName(fieldReference), fieldValue.getText(), project);
+    }
+  }
+
+  @Nullable
+  private static GoCompositeLit getStructLiteralByReference(@NotNull GoReferenceExpression fieldReference,
+                                                            @NotNull GoAssignmentStatement assignment) {
+    PsiElement previousElement = PsiTreeUtil.skipSiblingsBackward(assignment, PsiWhiteSpace.class);
+    if (previousElement instanceof GoSimpleStatement) {
+      return getStructLiteral(fieldReference, (GoSimpleStatement)previousElement);
+    }
+    if (previousElement instanceof GoAssignmentStatement) {
+      return getStructLiteral(fieldReference, (GoAssignmentStatement)previousElement);
+    }
+    return null;
+  }
+
+  @Nullable
+  private static GoCompositeLit getStructLiteral(@NotNull GoReferenceExpression fieldReference,
+                                                 @NotNull GoSimpleStatement structDeclaration) {
+    GoShortVarDeclaration varDeclaration = structDeclaration.getShortVarDeclaration();
+    if (varDeclaration == null) return null;
+    GoVarDefinition structVarDefinition =
+      ContainerUtil.find(varDeclaration.getVarDefinitionList(),
+                         definition -> GoPsiImplUtil.getName(definition).equals(getQualifierName(fieldReference)));
+    return structVarDefinition != null && isStructWithField(fieldReference, structVarDefinition) ?
+           ObjectUtils.tryCast(structVarDefinition.getValue(), GoCompositeLit.class) : null;
+  }
+
+  @Nullable
+  private static GoCompositeLit getStructLiteral(@NotNull GoReferenceExpression fieldReference,
+                                                 @NotNull GoAssignmentStatement structAssignment) {
+    GoExpression struct = ContainerUtil.find(structAssignment.getLeftHandExprList().getExpressionList(),
+                                             expression -> isStructByReference(tryGetReference(expression), fieldReference));
+    return struct != null ? ObjectUtils.tryCast(GoPsiImplUtil.getExpressionValue(structAssignment, struct), GoCompositeLit.class) : null;
+  }
+
+  private static boolean isStructByReference(@Nullable GoReferenceExpression structReference,
+                                             @NotNull GoReferenceExpression fieldReference) {
+    PsiElement structDefinition = structReference != null ? structReference.resolve() : null;
+    GoVarDefinition structVarDefinition = ObjectUtils.tryCast(structDefinition, GoVarDefinition.class);
+    return structVarDefinition != null &&
+           GoPsiImplUtil.getName(structVarDefinition).equals(getQualifierName(fieldReference)) &&
+           isStructWithField(fieldReference, structVarDefinition);
+  }
+
+  @Contract("null, _, _ -> false")
+  private static boolean isUninitializedFieldReferenceByQualifier(@Nullable GoExpression expression,
```

Comment:
split it method in according in SRP. You give `qaulifierName` only for check `qualifierName.equals(getQualifierName(fieldReference))`